models/base/scratch/bigquery/snowplow_media_player_base_events_this_run.sql

```diff
@@ -0,0 +1,305 @@
+{{
+  config(
+    tags=["this_run"]
+  )
+}}
+
+{%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model(ref('snowplow_media_player_base_sessions_this_run'),
+                                                                            'start_tstamp',
+                                                                            'end_tstamp') %}
+
+-- handling relations for integration tests
+{% if target.schema.startswith('gh_sp_media_player_dbt_') or 'dbt_matus' in target.schema %}
+  {%- set relation = ref('snowplow_media_player_events_stg') %}
+{% else %}
+  {%- set relation = source('atomic', 'events') %}
+{% endif %}
+
+with prep as (
+
+  select
+    *,
+    dense_rank() over (partition by ev.domain_sessionid order by ev.start_tstamp) AS event_in_session_index,
+
+  from (
+    -- without downstream joins, it's safe to dedupe by picking the first event_id found.
+    select
+      array_agg(e order by e.collector_tstamp limit 1)[offset(0)].*
+
+    from (
```

Comment:
Bigquery actually supports `qualify` now so you can do this the same way we do in snowflake+databricks which looks nicer